docs/core-features/comparisons/clipping.md

```diff
@@ -0,0 +1,214 @@
+# Clipping
+
+This document explains the clipping strategy for comparison operations and demonstrates two implementation methods.
+
+## Introduction
+
+The clipping strategy optimizes the required bit width of the subtraction strategy - it clips the bigger operand and then does the subtraction with fewer bits.
+
+For example, the results of `compare(3, 40)` and `compare(3, 4)` are the same,  so clipping the bigger operand before subtraction reduces the bit width needed.
+
+### Requirements
+
+The clipping method clips the bigger operand and then does the subtraction with fewer bits:
+
+1. The clipping requires the operands to have **different** bit widths.
+2. The subtraction then requires the operands to have the **same** bit widths.
+
+For example,  when comparing `uint3` and `uint6`, we need to convert both of them to `uint4` in some way to do the subtraction and proceed with the Table Lookup (TLU) in 4 bits. In other words:
+
+```python
+x.bit_width != y.bit_width
+```
+
+```python
+smaller = x if x.bit_width < y.bit_width else y
+bigger = x if x.bit_width > y.bit_width else y
+clipped = lambda value: np.clip(value, smaller.min() - 1, smaller.max() + 1)
+any(
+    (
+        bit_width <= MAXIMUM_TLU_BIT_WIDTH and
+        bit_width <= bigger.dtype.bit_width and
+        bit_width > smaller.dtype.bit_width
+    )
+    for bit_width in [
+        (smaller - clipped(bigger)).bit_width,
+        (clipped(bigger) - smaller).bit_width,
+    ]
+  )
+```
+
+There are 2 methods to implement clipping.
+
+## 1. Three TLUs: bigger clipped and smaller cast
+
+### How it works
+
```

Comment:
I stop here my review, I will speak offline with the people: those docs look
- complicated
- maybe too long / too precise
to me. In my opinion, we try to explain things which are very complicated for the final user, so 99% of them will not understand and we'll lose them. 

I'll discuss with @BourgerieQuentin @rudy-6-4 about that. So maybe @yuxizama we'll make you wait a bit, sorry